fanficfare/adapters/adapter_novelfull.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import absolute_import
+import logging
+import re
+
+# py2 vs py3 transition
+from ..six.moves.urllib import parse as urlparse
+
+from .base_adapter import BaseSiteAdapter
+
+
+logger = logging.getLogger(__name__)
+
+
+def getClass():
+    return NovelFullSiteAdapter
+
+
+class NovelFullSiteAdapter(BaseSiteAdapter):
+    @staticmethod
+    def getSiteDomain():
+        return "novelfull.com"
+
+    def getSiteURLPattern(self):
+        return r"https?://%s/(?P<name>.+).html?" % re.escape(self.getSiteDomain())
+
+    def extractChapterUrlsAndMetadata(self):
+        logger.debug("URL: %s", self.url)
+
+        data = self.get_request(self.url)
+
+        soup = self.make_soup(data)
+
+        self.story.setMetadata("title", soup.select_one("h3.title").text)
+        self.story.setMetadata(
+            "author",
+            " ".join(
+                s.text for s in soup.find("h3", text="Author:").fetchNextSiblings()
+            ),
+        )
+        self.story.setMetadata("authorId", self.story.getMetadata("author"))
+
+        status = soup.find("a", href=re.compile("status")).text
+
+        if status == "Completed":
+            self.story.setMetadata("status", "Completed")
+
+        cover_url = soup.find("div", class_="book").find("img").attrs["src"]
+        self.setCoverImage(self.url, urlparse.urljoin(self.url, cover_url))
+
+        self._crawl_chapters(self.url)
+
+        self.setDescription(
+            self.url, soup.select_one("div.desc-text").get_text(separator="\n")
```

Comment:
setDescription() can take HTML, no need to convert to text here.

Reply:
Happy to make this change, when confirming the functionality with `-o keep_summary_html=False` I noticed that paragraph tags with no whitespace separation will get jammed together in the produced description e.g. https://novelfull.com/perfect-secret-love-the-bad-new-wife-is-a-little-sweet.html gets the description 

> “This guy, how heavy is his taste, he’s still able to eat?” Waking up, she looked at her reflection in the mirror, explosive hair, tattoos, and a demon-like face. Look at her for more than a second and you’ll have spicy eyes (your eyes will bleed-aka she looks really ugly).Before her rebirth, she was in love with someone else, bent on escaping, and after having relations with him, hates him deep to his bones.After her rebirth, she looked at the beauty on the bed, seriously thinking, the one who left his shadow in the past, seemingly should be him?In her past life, her mind was muddled. She tried to get rid of the outstandingly beautiful husband that she didn’t want, was victimized by slag men and cheap women, and her most trusted friend brainwashed her. In the end, she found people rebelling and friends deserting (isolated and alone).In this lifetime, all of the evil people scheming and longing for her divorce should yield. Sorry but this young miss’s IQ is on the line!Translator: eunimon_Editor: Caron_

Not something to worry about in this PR though